src/BarrelPlanarMPGDTracker_geo.cpp

```diff
@@ -283,14 +291,20 @@ static Ref_t create_BarrelPlanarMPGDTracker_geo(Detector& description, xml_h e,
         string module_name = _toString(module, "module%d");
         DetElement mod_elt(lay_elt, module_name, module);
         double mod_z       = 0.5 * dimensions.length();
-        double z_placement = mod_z - j * nz * mod_z; // z location for module placement
+        double z_placement = mod_z - 0.5 * pcb_feb_ext -
+                             j * (nz * mod_z - pcb_feb_ext); // z location for module placement
```

Comment:
Even the original expression looks a bit fishy to me. I doubt it works for nz != 2. Maybe, it should have `2` instead of `nz`, because mod_z is half-length.